src/components/AddItemForm.js

```diff
@@ -0,0 +1,82 @@
+import { useState } from 'react';
+import { db } from '../lib/firebase';
+import { addDoc, collection } from 'firebase/firestore';
+
+const AddItemForm = () => {
+  const [timeframe, setTimframe] = useState('7');
+  const [newItem, setNewItem] = useState('');
+
+  const handleSelect = (e) => {
+    setTimframe(e.target.value);
+  };
+
+  const handleItemAdd = (e) => {
+    setNewItem(e.target.value);
+  };
+  const addItem = async (
+    newItem,
+    timeframe,
+    lastPurchased = null,
+    token = 'reda and tobi',
+  ) => {
+    const ListRef = collection(db, 'List1');
+    await addDoc(ListRef, {
+      Item: newItem,
+      timeframe: parseInt(timeframe),
+      lastPurchased,
+      token,
+    });
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    addItem(newItem, timeframe);
+  };
+  return (
+    <form onSubmit={handleSubmit}>
+      <label htmlFor="newItem">Item name:</label>
+      <input
+        type="text"
+        id="newItem"
+        value={newItem}
+        onChange={handleItemAdd}
+        name="newItem"
+        placeholder="Add Item"
+        required
+      />
+      <fieldset>
+        <legend>How soon will you buy this again ?</legend>
+        <input
+          type="radio"
+          id="soon"
+          value="7"
+          checked={timeframe === '7'}
+          onChange={handleSelect}
+          name="timeframe"
+        />
+        <label htmlFor="soon">Soon</label>
+        <input
+          type="radio"
+          id="kind-of-soon"
+          value="14"
+          checked={timeframe === '14'}
+          onChange={handleSelect}
+          name="timeframe"
+        />
+        <label htmlFor="kind-of-soon">kind of soon</label>
+        <input
+          type="radio"
+          id="not-soon"
+          value="30"
+          checked={timeframe === '30'}
+          onChange={handleSelect}
+          name="timeframe"
+        />
+        <label htmlFor="not-soon">not-Soon</label>
```

Comment:
Not very important, but you can write "Not Soon" here to more closely follow the wireframe :slightly_smiling_face: https://github.com/the-collab-lab/smart-shopping-list/blob/main/_resources/wireframes/add-item.png